x/auth/vesting/types/vesting_account_test.go

```diff
@@ -578,6 +578,152 @@ func TestTrackUndelegationPeriodicVestingAcc(t *testing.T) {
 	require.Equal(t, sdk.Coins{sdk.NewInt64Coin(stakeDenom, 25)}, pva.DelegatedVesting)
 }
 
+func TestGetVestedCoinsPermLockedVestingAcc(t *testing.T) {
+	now := tmtime.Now()
+	endTime := now.Add(1000 * 24 * time.Hour)
+
+	_, _, addr := testdata.KeyTestPubAddr()
+	origCoins := sdk.Coins{sdk.NewInt64Coin(feeDenom, 1000), sdk.NewInt64Coin(stakeDenom, 100)}
+	bacc := authtypes.NewBaseAccountWithAddress(addr)
+
+	// require no coins are vested until schedule maturation
+	plva := types.NewPermanentLockedVestingAccount(bacc, origCoins)
+	vestedCoins := plva.GetVestedCoins(now)
+	require.Nil(t, vestedCoins)
+
+	// require no coins be vested at end time
+	vestedCoins = plva.GetVestedCoins(endTime)
+	require.Nil(t, vestedCoins)
+}
+
+func TestGetVestingCoinsPermLockedVestingAcc(t *testing.T) {
+	now := tmtime.Now()
+	endTime := now.Add(1000 * 24 * time.Hour)
+
+	_, _, addr := testdata.KeyTestPubAddr()
+	origCoins := sdk.Coins{sdk.NewInt64Coin(feeDenom, 1000), sdk.NewInt64Coin(stakeDenom, 100)}
+	bacc := authtypes.NewBaseAccountWithAddress(addr)
+
+	// require all coins vesting at the beginning of the schedule
+	plva := types.NewPermanentLockedVestingAccount(bacc, origCoins)
+	vestingCoins := plva.GetVestingCoins(now)
+	require.Equal(t, origCoins, vestingCoins)
+
+	// require all coins vesting at the end time
+	vestingCoins = plva.GetVestingCoins(endTime)
+	require.Equal(t, origCoins, vestingCoins)
+}
+
+func TestSpendableCoinsPermLockedVestingAcc(t *testing.T) {
+	now := tmtime.Now()
+	endTime := now.Add(1000 * 24 * time.Hour)
+
+	_, _, addr := testdata.KeyTestPubAddr()
+	origCoins := sdk.Coins{sdk.NewInt64Coin(feeDenom, 1000), sdk.NewInt64Coin(stakeDenom, 100)}
+	bacc := authtypes.NewBaseAccountWithAddress(addr)
+
+	// require that all coins are locked in the beginning of the vesting
+	// schedule
+	plva := types.NewPermanentLockedVestingAccount(bacc, origCoins)
+	lockedCoins := plva.LockedCoins(now)
+	require.True(t, lockedCoins.IsEqual(origCoins))
+
+	// require that all coins are still locked at end time
+	lockedCoins = plva.LockedCoins(endTime)
+	require.True(t, lockedCoins.IsEqual(origCoins))
+
+	// receive some coins
+	// require that only received coins are spendable since the original coins
+	// are all locked
+	lockedCoins = plva.LockedCoins(endTime)
```

Comment:
This seems like an inaccurate assertion. There aren't any "coins received" in the test case. Same goes for all the `TestSpendableCoins{SomeOther}VestingAcc()` implementations though. Let me know if i'm missing something here, or if we should actually test for receiving some coins. This would require setting up a bank keeper iiuc?

Reply:
I had a look at the other Spendable tests, the "receive some coins" part always seems to just be a copy/pate of the previous line... I'm a bit confused here too, maybe @alexanderbez has an idea?

Reply:
So these comments are remnants from ADR-004 implementation with the test modifications [here](https://github.com/cosmos/cosmos-sdk/pull/5572/files#diff-79cdc833ea7de9c9dcb705ff51d3b0d2648ce639a7438c85e5a8cea2671577b1). Looks like I forgot to remove the `// receive some coins` comments :-/

Apologies.

Reply:
Before ADR-004, these tests did have accounts receive coins, that's why those comments existed. They should be removed now 👍